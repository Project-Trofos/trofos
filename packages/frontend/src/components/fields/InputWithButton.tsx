import React, { useState } from 'react';
import { Space, Input, Button } from 'antd';

type InputWithButtonProps = {
  handleClick: (value: string) => void;
  buttonText?: string;
  inputPlaceholder?: string;
};

export default function InputWithButton({ handleClick, buttonText, inputPlaceholder }: InputWithButtonProps) {
  const [value, setValue] = useState('');
  const [isError, setIsError] = useState(false);

  return (
    <Space style={{ display: 'flex' }}>
      <Input
        onPressEnter={() => handleClick(value)}
        value={value}
        onChange={(e) => setValue(e.target.value)}
        placeholder={inputPlaceholder}
        status={isError ? 'error' : undefined}
      />

      <Button
        htmlType="submit"
        onClick={() => {
          setIsError(!value);
          if (value) {
            handleClick(value);
          }
        }}
      >
        {buttonText}
      </Button>
    </Space>
  );
}
