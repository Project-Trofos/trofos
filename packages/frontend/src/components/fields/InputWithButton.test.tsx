import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import InputWithButton from './InputWithButton';

describe('test InputWithButton', () => {
  describe('when rendering', () => {
    const BUTTON_TEXT = 'ok';
    const PLACEHOLDER = 'placeholder';

    const setup = () => {
      const handleClickFunc = vi.fn();
      const utils = render(
        <InputWithButton handleClick={handleClickFunc} buttonText={BUTTON_TEXT} inputPlaceholder={PLACEHOLDER} />,
      );

      return { ...utils, handleClickFunc };
    };

    it('should render correct fields', () => {
      const { baseElement } = setup();
      // Ensure button is present
      expect(screen.getByText(BUTTON_TEXT)).toBeInTheDocument();

      // Ensure input are present
      expect(screen.getByPlaceholderText(PLACEHOLDER)).toBeInTheDocument();

      // Compare with snapshot to ensure structure remains the same
      expect(baseElement).toMatchSnapshot();
    });

    it('should call function with correct value when button is clicked', () => {
      const { handleClickFunc } = setup();
      const VALUE = 'VALUE';
      const button = screen.getByText(BUTTON_TEXT);
      const input = screen.getByPlaceholderText(PLACEHOLDER);

      fireEvent.change(input, { target: { value: VALUE } });
      fireEvent.click(button);

      expect(handleClickFunc).toHaveBeenCalledTimes(1);
      expect(handleClickFunc).toHaveBeenCalledWith(VALUE);
    });
  });
});
