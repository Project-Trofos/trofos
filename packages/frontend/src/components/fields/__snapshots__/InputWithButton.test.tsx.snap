// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test InputWithButton when rendering should render correct fields 1`] = `
<body>
  <div>
    <div
      class="ant-space css-dev-only-do-not-override-1qhh31y ant-space-horizontal ant-space-align-center"
      style="display: flex;"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <input
          class="ant-input css-dev-only-do-not-override-1qhh31y"
          placeholder="placeholder"
          type="text"
          value=""
        />
      </div>
      <div
        class="ant-space-item"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1qhh31y ant-btn-default"
          type="submit"
        >
          <span>
            ok
          </span>
        </button>
      </div>
    </div>
  </div>
</body>
`;
